src/main.js

```diff
@@ -127,6 +135,68 @@ class MiniGraphCard extends LitElement {
     }
   }
 
+  checkSections() {
+    let root = document.querySelector('home-assistant');
+    root = root && root.shadowRoot;
+    root = root && root.querySelector('home-assistant-main');
+    root = root && root.shadowRoot;
+    root = root && root.querySelector('ha-drawer');
+    root = root && root.querySelector('partial-panel-resolver');
+    root = root && root.shadowRoot || root;
+    root = root && root.querySelector('ha-panel-lovelace');
+    root = root && root.shadowRoot;
+    root = root && root.querySelector('hui-root');
+    if (root) {
+      const ll = root.lovelace;
+      ll.current_view = root.___curView;
+      const panelType = ll.rawConfig.views[ll.current_view].type;
+      return panelType === 'sections';
+    }
+    return false;
+  }
+
+  getCurrentLayout() {
+    const layout = this.getLayoutOptions();
+    const layoutConfigured = this.config.layout_options !== undefined;
+    const columns = Math.max(layoutConfigured
+      ? this.config.layout_options.grid_columns : layout.grid_columns, layout.grid_min_columns);
+    const rows = Math.max(layoutConfigured
+      ? this.config.layout_options.grid_rows : layout.grid_rows, layout.grid_min_rows);
+    return { grid_columns: columns, grid_rows: rows };
+  }
+
+  getLayoutSize(layout) {
+    if (!this.sections) {
```

Comment:
Reduce? 

```js
return this.sections && layout.grid_rows === 2 ? 'small' : '';
```

And if that is enough, maybe remove the function and do the logic in the render directly... (Because it seems used only once.)